Source/Parallelization/CurrentSynchronize.H

```diff
@@ -0,0 +1,171 @@
+/* Copyright 2019 Axel Huebl, Weiqun Zhang
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_CURRENTSYNCHRONIZE_H
+#define WARPX_CURRENTSYNCHRONIZE_H
+
+#include <AMReX_Array4.H>
+#include <AMReX_REAL.H>
+#include <AMReX_BLassert.H>
+#include <AMReX_Extension.H>
+#include <AMReX_GpuQualifiers.H>
+
+
+/** Fill a current coarse patch with averaged values from a fine patch
+ *
+ * Fills the values of the current for a selected component on the coarse patch
+ * by averaging the values of the current of the fine patch.
+ *
+ * @tparam IDim j-field component on which the averaging is performed
+ */
+template< int IDim >
+class WarpxSyncCurrent
+{
+public:
+    /** Construct with fine and coarse patch and their refinement ratio
+     *
+     * @param[in]     fine read-only fine patch
+     * @param[in,out] coarse overwritten coarse patch
+     * @param[in]     refinement_ratio ratio between coarse and fine patch granularity
+     *                (currently, only a value of is implemented)
+     */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    WarpxSyncCurrent(
+        amrex::Array4<amrex::Real const> const fine,
+        amrex::Array4<amrex::Real      > const coarse,
```

Comment:
Do you think it matters whether we pass these by value or `const &` here?

Reply:
I think copy elision will guarantee it is moved into the member anyway, no?
How expensive is that host-side copy into a member actually?

Reply:
slack discussion: `Array4` is the container object we copy, so this is fine.
`std::move` added.